include/power/pm_policy.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _PM_POLICY_H_
+#define _PM_POLICY_H_
+
+#include <power/power.h>
+
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Define the base struct necessary for a pm_policy.
+ *
+ * Different policies can have their own struct type according with
+ * their necessity, the only requirements is the struct contains this
+ * macro in the beginning of the declaration and be statically defined
+ * using @c PM_POLICY_DEFINE.
+ */
+#define PM_POLICY_BASE \
+	struct { \
+		const char *name; \
+		const size_t len; \
+		enum power_states (*next_state_get)(struct pm_policy *policy,\
```

Comment:
I think the first argument should be `const struct pm_policy *`.  There should be no reason to have mutable data in the policy structure: if it needs state, it should store it somewhere else.

Reply:
I was in doubt about just adding a pointer that can be used by the implementation to store its context or allow extending the type. I can change it, in the end it will not increase the struct size because I can get rid of `len` and it will simplify some things.